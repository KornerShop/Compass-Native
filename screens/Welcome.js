import React, {Component} from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {View, Text, StyleSheet} from 'react-native'
import {ButtonGroup} from 'react-native-elements'

import {setLanguagePreference} from '../redux/actions/actions'

class Welcome extends Component {
  constructor(props) {
    super(props)
  }
  // Updates language preference, triggering the tabbed UI to render
  updateIndex(idx) {
    // Dispatch action to change the language key on our redux store,
    // which would hopefully update the 'selected' on line 26
    // if it doesn't, let's talk
    const language = idx === 1 ? 'es' : 'en'
    this.props.setLanguagePreference(language)
    this.props.toggleStart()
  }
  render() {
    return (
      <ButtonGroup
        onPress={this.updateIndex.bind(this)}
        selectedIndex={this.props.language === 'en' ? 0 : 1}
        buttons={['English', 'EspaÃ±ol']}
        containerStyle={{height: 150}}
      />
    )
  }
}

const styles = StyleSheet.create({})

const mapStateToProps = ({language}) => {
  language
}

const mapDispatchToProps = dispatch => {
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch)
}

export default connect()(Welcome)
