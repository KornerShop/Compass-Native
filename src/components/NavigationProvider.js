import React from 'react'
import {
  NavigationProvider,
  TabNavigation,
  StackNavigation,
} from '@expo/ex-navigation'
import Router from '../navigation/Router'

export default () =>
  <NavigationProvider router={Router}>
    <StackNavigation initialRoute={Router.getRoute('rootNavigation')} />
  </NavigationProvider>
