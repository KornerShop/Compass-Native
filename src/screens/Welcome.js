import React, {Component} from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {
  Image,
  View,
  Text,
  StyleSheet,
  AsyncStorage,
  Dimensions,
  ActivityIndicator,
  StatusBar,
} from 'react-native'
import {FontAwesome} from '@expo/vector-icons'
import {ButtonGroup} from 'react-native-elements'
import styled from 'styled-components/native'

import {setLanguagePreference} from '../redux/actions/actions'

// update office state
// custom marker views
// Modal: update location state; update office state
// translations
// submit button on form
// lock in form style
// eligible and ineligible screens
// test on Android
// uncomment out get AsyncStorage in Main.js
// take stock of issues, re-organize via waffle
// organize code
// tests
// look into reprecussions of ejecting w/ regard to pushing to app stores
// smile - we can do it!

const ActivityIndicatorWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

const BackgroundImage = props =>
  <Image>
    {props.children}
  </Image>

const WelcomeUIContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-content: center;
`

const Logo = styled.Text`
  text-align: center;
  font-size: 65;
  font-weight: 300;
  color: white;
  background-color: transparent;
  margin-bottom: 30
`

class Welcome extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoading: true,
      selectedLanguage: 2,
    }
  }

  async updateIndex(idx) {
    this.setState({selectedLanguage: idx})
    const language = idx === 1 ? 'es' : 'en'
    try {
      await AsyncStorage.setItem('language', language)
    } catch (error) {
      console.warn(error)
    }
    this.props.setLanguagePreference(language)
    this.props.toggleStart()
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <StatusBar barStyle="light-content" />
        <Image
          onLoad={() => this.setState({isLoading: false})}
          source={require('../assets/shopper.png')}
          tintColor={'red'}
          style={{
            flex: 1,
            height: this.props.orientation.height,
            width: this.props.orientation.width,
            resizeMode: 'cover',
          }}>
          {this.state.isLoading
            ? <ActivityIndicatorWrapper>
                <ActivityIndicator color="tomato" size="large" />
              </ActivityIndicatorWrapper>
            : <WelcomeUIContainer>
                <Logo>
                  C<FontAwesome name="compass" size={50} color="white" />mpass
                </Logo>
                <ButtonGroup
                  onPress={this.updateIndex.bind(this)}
                  selectedIndex={this.state.selectedLanguage}
                  buttons={['English', 'EspaÃ±ol']}
                  textStyle={{color: 'white', fontSize: 18, fontWeight: 'bold'}}
                  selectedTextStyle={{color: 'slategray'}}
                  innerBorderStyle={{
                    color: 'white',
                    width: 3,
                  }}
                  containerStyle={{
                    height: 60,
                    marginTop: 30,
                    backgroundColor: 'transparent',
                    borderWidth: 3,
                    borderRadius: 7,
                    borderColor: 'white',
                    width: 300,
                    alignSelf: 'center',
                  }}
                  selectedBackgroundColor="white"
                />
              </WelcomeUIContainer>}
        </Image>
      </View>
    )
  }
}

const mapStateToProps = ({orientation}) => ({orientation})

const mapDispatchToProps = dispatch => ({
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch),
})

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
