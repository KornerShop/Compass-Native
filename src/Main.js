import React from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'

import {
  AsyncStorage,
  ActivityIndicator,
  View,
  StyleSheet,
  Dimensions,
} from 'react-native'

import NavigationProvider from './components/NavigationProvider'
import Welcome from './screens/Welcome'

import {updateOrientation} from './redux/actions/actions'
import {setLanguagePreference} from './redux/actions/actions'

class Main extends React.Component {
  constructor() {
    super()
    this.state = {
      started: false,
      isLoading: true,
    }
  }
  toggleStart() {
    this.setState({started: !this.state.started})
  }
  async componentWillMount() {
    // Not ideal, but this action is necessary for getting the initial orientation of the device
    this.props.updateOrientation(Dimensions.get('window'))
    try {
      const language = await AsyncStorage.getItem('language')
      this.setState({
        isLoading: false,
      })
      if (language) {
        this.setState({
          started: true,
        })
        setLanguagePreference(language)
      }
    } catch (error) {
      console.warn(error)
    }
  }
  renderRoot(Component) {
    return (
      <View
        style={{flex: 1}}
        onLayout={() => this.props.updateOrientation(Dimensions.get('window'))}>
        <Component toggleStart={this.toggleStart.bind(this)} />
      </View>
    )
  }
  render() {
    if (this.state.isLoading) {
      return (
        <View style={styles.loader}>
          <ActivityIndicator color="tomato" size="large" />
        </View>
      )
    }
    return this.state.started
      ? this.renderRoot(NavigationProvider)
      : this.renderRoot(Welcome)
  }
}

const mapDispatchToProps = dispatch => ({
  updateOrientation: bindActionCreators(updateOrientation, dispatch),
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch),
})

const styles = StyleSheet.create({
  loader: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'papayawhip',
  },
})

export default connect(null, mapDispatchToProps)(Main)
