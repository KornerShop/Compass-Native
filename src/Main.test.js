import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';

import Main, { Unwrapped as UnwrappedMain } from './Main';
import Welcome from './screens/Welcome';
import NavigationProvider from './containers/NavigationProvider';

const wait = (exp, ms) =>
  new Promise(resolve => setTimeout(() => resolve(exp), ms));

describe('shallow tests', () => {
  test('UnwrappedMain renders correctly', () => {
    const wrapper = shallow(<UnwrappedMain />);
    expect(wrapper.state()).toMatchObject({
      started: false,
      isLoading: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  // test('Main renders correctly', () => {
  //   const store = mockStore(initState);
  //   const wrapper = shallow(
  //     <Provider store={store}>
  //       <Main />
  //     </Provider>
  //   );
  //   expect(wrapper).toMatchSnapshot();
  // });

  test('Main renders Welcome after initial load', async () => {
    const wrapper = shallow(<UnwrappedMain />);
    await wait(wrapper.setState({ isLoading: false }), 100);
    expect(wrapper.find(Welcome)).toBeTruthy();
  });

  test('Main renders navigationProvider after language chosen', async () => {
    const wrapper = shallow(<UnwrappedMain />);
    await wait(wrapper.setState({ isLoading: false }), 100);
    wrapper.setState({ started: true });
    expect(wrapper.find(NavigationProvider)).toBeTruthy();
  });

  test('Main renders navigationProvider after language chosen (event)', async () => {
    const wrapper = shallow(<UnwrappedMain />);
    await wait(wrapper.setState({ isLoading: false }), 100);
    wrapper.find(Welcome).simulate('press', [1])
    expect(wrapper.find(NavigationProvider)).toBeTruthy();
  });
});

// tests for content ('compass', button on welcome, spanish if language Spanish elsewhere)
// make checklist after completing this file ^
// test to make sure mapdispatchtoprops functions are present
// redux tests (Redux guide) + props tests for connected export (perhaps related)
// see 'Main renders correctly', above...needs mock store & provider (see btholt's file)
