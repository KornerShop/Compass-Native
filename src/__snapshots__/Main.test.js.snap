// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer tests UnwrappedMain renders correctly 1`] = `
<Styled(View)>
  <ActivityIndicator
    animating={true}
    color="#00897b"
    hidesWhenStopped={true}
    size="large"
  />
</Styled(View)>
`;
